elastic/apm-server
# sourcemap: strip carriage returns in sourcemap.Map

## Motivation/summary

If the source is formatted using Windows line endings, the carriage returns will be included in the stored source content. This leads to incorrect source code context rendering in Kibana: https://github.com/elastic/kibana/issues/80747

Instead of just splitting on newlines, use [bufio.Scanner](https://golang.org/pkg/bufio/#Scanner) which will strip one carriage return preceding the newline character. As a side benefit, this should be slightly more efficient as it means we don't need to split the whole file at once only to capture 11 lines of source code.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

See https://github.com/elastic/kibana/issues/80747#issuecomment-716292425

1. Create a small web app with some Javascript that will throw. Save the source code with Windows line endings (in vim: `set ff=dos`)
2. Use webpack to create a minified version with a sourcemap.
3. Run server with RUM enabled, upload sourcemap, run the web app and wait for an error to be indexed.
4. Navigate to the error, check that the source context renders as expected.

## Related issues

#4347
